apps/desktop-wallet/src/App.tsx

```diff
@@ -189,26 +194,76 @@ const App = () => {
 
               restorePendingTransactions(mempoolTxHashes, storedPendingTxs)
             })
-          dispatch(syncAddressesHistoricBalances())
+
+          dispatch(syncAddressesAlphHistoricBalances())
         }
       } else if (addressesStatus === 'initialized') {
-        if (!isTokensMetadataUninitialized && !isLoadingTokensMetadata && newUnknownTokens.length > 0) {
+        if (areVerifiedFungibleTokensInitialized && !isLoadingUnverifiedFungibleTokens && newUnknownTokens.length > 0) {
           dispatch(syncUnknownTokensInfo(newUnknownTokens))
         }
       }
     }
   }, [
     addressHashes.length,
     addressesStatus,
-    assetsInfo.status,
+    areVerifiedFungibleTokensInitialized,
     dispatch,
+    isLoadingUnverifiedFungibleTokens,
     isSyncingAddressData,
-    isLoadingTokensMetadata,
-    isTokensMetadataUninitialized,
     network.status,
     newUnknownTokens
   ])
 
+  // Fetch verified tokens from GitHub token-list and sync current and historical prices for each verified fungible
+  // token found in each address
+  useEffect(() => {
+    if (network.status === 'online' && !isLoadingVerifiedFungibleTokens) {
+      if (fungibleTokens.status === 'uninitialized') {
+        dispatch(syncVerifiedFungibleTokens())
+      } else if (verifiedFungibleTokenSymbols.uninitialized.length > 0) {
+        const symbols = verifiedFungibleTokenSymbols.uninitialized
+
+        dispatch(syncTokenCurrentPrices({ verifiedFungibleTokenSymbols: symbols, currency: settings.fiatCurrency }))
+        dispatch(syncTokenPriceHistories({ verifiedFungibleTokenSymbols: symbols, currency: settings.fiatCurrency }))
+      }
+    }
+  }, [
+    dispatch,
+    fungibleTokens.status,
+    isLoadingVerifiedFungibleTokens,
+    network.status,
+    settings.fiatCurrency,
+    verifiedFungibleTokenSymbols.uninitialized
+  ])
+
+  useEffect(() => {
+    if (
+      network.status === 'online' &&
+      !isLoadingVerifiedFungibleTokens &&
+      verifiedFungibleTokenSymbols.uninitialized.length > 1
+    ) {
+      console.log(
+        'TODO: Sync address verified tokens balance histories for',
+        verifiedFungibleTokenSymbols.uninitialized.filter((symbol) => symbol !== ALPH.symbol)
+      )
+    }
+  }, [isLoadingVerifiedFungibleTokens, network.status, verifiedFungibleTokenSymbols.uninitialized])
+
+  const refreshTokensLatestPrice = useCallback(() => {
+    dispatch(
+      syncTokenCurrentPrices({
+        verifiedFungibleTokenSymbols: verifiedFungibleTokenSymbols.withPriceHistory,
+        currency: settings.fiatCurrency
+      })
+    )
+  }, [dispatch, settings.fiatCurrency, verifiedFungibleTokenSymbols.withPriceHistory])
+
+  useInterval(
+    refreshTokensLatestPrice,
+    60000,
+    network.status !== 'online' || verifiedFungibleTokenSymbols.withPriceHistory.length === 0
+  )
```

Comment:
This is how I re-implemented the polling mechanism that was in place using **RTK Query** before.

Reply:
Good idea 👍